### DISCUSSION TITLE
How to sub 1 year?

### DISCUSSION THREAD

**Discussion by numfin** [time-rs/time#577] [Question] [answered]
How to sub 1 year from OffsetDateTime? 
```rs
let extra_day = if current_year() % 4 === 0 { 1 } else { 0 }; 
days(365 + extra_day);
```
 is not a very clean solution :D

**Comment by numfin** [comment, marked as answer]
this one works too i guess
```rs
let now = time::OffsetDateTime::now_utc();
let sub_year = now.replace_year(now.year() - 1);
```
but `sub_year/day/month/etc()` will be much cleaner

**Reply by jhpratt** [reply]
Replacing the year would likely be the idiomatic way to do it at the moment. I won't be adding dedicated methods for this, as the overhead isn't worth it.
